core/payment/src/api/allocations.rs

```diff
@@ -32,187 +33,252 @@ const DEFAULT_TESTNET_NETWORK: NetworkName = NetworkName::Holesky;
-                .map_err(|err| bad_req_and_log(format!("Validate token failed (1): {err}")))?;
-            log::debug!("Selected network {}-{}-{}", driver, network, token);
-            format!("{}-{}-{}", driver, network, token)
-        } else {
-            let default_token = get_default_token(&driver, &network);
+            };
+            Ok(platform)
+        }
 
-            log::debug!(
-                "Selected network with default token {}-{}-{}",
+        pub fn from_payment_platform_str(
+            payment_platform_str: &str,
+        ) -> Result<PaymentPlatformTriple, HttpResponse> {
+            // payment_platform is of the form driver-network-token
+            // eg. erc20-rinkeby-tglm
+            let [driver_str, network_str, token_str]: [&str; 3] = payment_platform_str
+                .split('-')
+                .collect::<Vec<_>>()
+                .try_into()
+                .map_err(|err| {
+                    bad_req_and_log(format!(
+                        "paymentPlatform must be of the form driver-network-token instead of {}",
+                        payment_platform_str
+                    ))
+                })?;
+
+            let network = validate_network(network_str)
+                .map_err(|err| bad_req_and_log(format!("Validate network failed (2): {err}")))?;
+
+            let driver = validate_driver(&network, driver_str)
+                .map_err(|err| bad_req_and_log(format!("Validate driver failed (2): {err}")))?;
+
+            let token = TokenName::from_token_string(&driver, &network, token_str)
+                .map_err(|err| bad_req_and_log(format!("Validate token failed (2): {err}")))?;
+
+            Ok(Self {
                 driver,
                 network,
-                default_token
-            );
-            format!("{}-{}-{}", driver, network, default_token)
+                token,
+            })
         }
-    };
-    Ok(platform_str)
-}
-
-fn payment_platform_validate_and_check(
-    payment_platform_str: &str,
-) -> Result<(DriverName, NetworkName, String), HttpResponse> {
-    // payment_platform is of the form driver-network-token
-    // eg. erc20-rinkeby-tglm
-    let [driver_str, network_str, token_str]: [&str; 3] = payment_platform_str
-        .split('-')
-        .collect::<Vec<_>>()
-        .try_into()
-        .map_err(|err| {
-            bad_req_and_log(format!(
-                "paymentPlatform must be of the form driver-network-token instead of {}",
-                payment_platform_str
-            ))
-        })?;
+    }
+    impl Display for PaymentPlatformTriple {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            write!(f, "{}-{}-{}", self.driver, self.network, self.token)
+        }
+    }
 
-    let network = validate_network(network_str)
-        .map_err(|err| bad_req_and_log(format!("Validate network failed (2): {err}")))?;
+    fn validate_network(network: &str) -> Result<NetworkName, String> {
```

Comment:
Common practice would be to use `anyhow` if you need to return something generic as string.